wordpress-mobile/WordPress-Android
# Convert `BlockProcessor` to kotlin

This converts `BlockProcessor.java` class to Kotlin.

- First, I added nullability annotations to `BlockProcessor` and the classes that use `BlockProcessor`.
- I broke the logic in `processBlock()` function, but then I corrected it with the commit, "Fix the broken logic in BlockProcessor.processBlock()"

-----

## To Test:



[test.webm](https://github.com/wordpress-mobile/WordPress-Android/assets/2471769/c92c8ecf-a458-4a2f-a5d8-95372c855706)

To test `BlockProcessor` class, you should follow the steps in the video for the different media types.
1. Edit a post.
2. Add an image.
3. Leave the post before the media upload is done.
4. Wait for uploading to be done.
5. Open the drafted post again.

-----

## Regression Notes

1. Potential unintended areas of impact

    - BlockProcessor class

6. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Tested manually

7. What automated tests I added (or what prevented me from doing so)

    - None

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
